src/Emogrifier/CssInliner.php

```diff
@@ -662,11 +663,28 @@ function (array $cssRule) {
      */
     private function removeUnmatchablePseudoComponents($selector)
     {
+        // The regex allows nested brackets via `(?2)`.
+        // A space is temporarily prepended because the callback can't determine if the match was at the very start.
+        $selectorWithNotsRemoved = \ltrim(\preg_replace_callback(
+            '/(\\s?+):not(\\([^\\(\\)]*+(?:(?2)[^\\(\\)]*+)*+\\))/i',
+            static function (array $matches) {
```

Comment:
I propose having a well-named regular method instead of the anonymous function and some well-named local variables to make the code more explicit.

Reply:
I've moved the anonymous function to a method, and documented the `$matches` parameter in PHPDoc.  Hopefully this makes it clear what they are without additionally assigning the values in `$matches` to local variables...?